### PR TITLE
close #94, close #95 - tests for Doctor::removeVisit and Doctor::appendVisit

### LINKED ISSUE #94: Doctor - removeVisit test

### LINKED ISSUE #95: Doctor - appendVisit test

### PR DESCRIPTION
**Pull request by mBialczak**
One review for two Issues:
1. Doctor - removeVisit test #94
2.  Doctor - appendVisit test #95

### CODE REVIEW

File: src/app/doctor/doctor.cpp [hunk 1]
```diff
@@ -14,7 +14,7 @@ void Doctor::appendVisit(const std::shared_ptr<Visit>& visit)
 
 void Doctor::removeVisit(const std::shared_ptr<Visit>& visit)
 {
-    visits_.erase(std::remove(visits_.begin(), visits_.end(), visit));
+    visits_.erase(std::remove(visits_.begin(), visits_.end(), visit), visits_.end());
```
Comment (fbtom) [inline src/app/doctor/doctor.cpp:17]: Are we sure, that the range should be erased here?

File: src/tests/ut/doctor/doctor_test.cpp [hunk 2]
```diff
@@ -1,17 +1,105 @@
...
+    auto first_expected_doctor = Clinic::getDoctors().at(0);
+    auto second_expected_doctor = Clinic::getDoctors().at(1);
+    auto third_expected_doctor = Clinic::getDoctors().at(2);
+    auto fourth_expected_doctor = Clinic::getDoctors().at(3);
+
+    EXPECT_EQ(Doctor::getDoctor("Jan", "Kowalski").front(), first_expected_doctor);
+    EXPECT_EQ(Doctor::getDoctor("Janusz", "Schmit").front(), second_expected_doctor);
+    EXPECT_EQ(Doctor::getDoctor("Gawel", "Pawel").front(), third_expected_doctor);
+    EXPECT_EQ(Doctor::getDoctor("Jan", "Poniatowski").front(), fourth_expected_doctor);
+}
```
Comment (fbtom) [inline src/tests/ut/doctor/doctor_test.cpp:30]: Two doctors would be enough ;)

File: src/tests/ut/doctor/doctor_test.cpp [hunk 3]
```diff
@@ -1,17 +1,105 @@
...
+
+    auto doctor_tested = Clinic::getDoctors().at(0);
+    auto irrelevant_doctor = Clinic::getDoctors().at(1);
+    createThreeSimpleVisits(irrelevant_doctor);
+    ASSERT_TRUE(doctor_tested->getVisits().empty());
+
+    const auto visit_to_append = irrelevant_doctor->getVisits().at(1);
+    doctor_tested->appendVisit(visit_to_append);
+    checkIfDoctorHasVisits(doctor_tested, {"second visit"});
+}
```
Comment (fbtom) [inline src/tests/ut/doctor/doctor_test.cpp:45]: Thinking of method Append use we should have a test in Visit which expects Visit to have doctor and doctor to have visit. 
It seems not necessary to have it here anyway. 
But as it's done let's keep it =)

**Review by fbtom — APPROVED** [review]

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

**Review by BaqWin — APPROVED** [review]

File: src/tests/ut/doctor/doctor_test.cpp (hunk 2 above)
Comment (mBialczak) [reply src/tests/ut/doctor/doctor_test.cpp:30]: Yeap. But there were four doctors already. I only decided that as if there are 4, then better to check them all.
All this test is strange a bit as it tests Visit and Clinic even more than Doctor class itself.

File: src/app/doctor/doctor.cpp (hunk 1 above)
Comment (mBialczak) [reply src/app/doctor/doctor.cpp:17]: I was surprised, but it proved to work. Previous version had segfaults.

**Comment by ziobron** [comment]
Your PR was merged!
🏅 2 XP granted. Thanks @mBialczak for making course materials better!

**Pull request merged**